### PR TITLE
Restart udevd in HA tests if HOSTNAME was changed

### PR DESCRIPTION
**Pull request by alvarocarvajald**
In SLES 15-SP3 when the hostname of the cluster nodes are changed (by test module `console/hostname`), drbd resources configured to the cluster fail to start (more info in [bsc#1177927](https://bugzilla.suse.com/show_bug.cgi?id=1177927#c15))

As explained in the bug report:
```
The reason of DRBD start failure by RA is because udev rules of creating DRBD
device (drbd_passive in this case) is not applied. Re-set the hostname(without a reboot)
is the reason of this issue, cause it makes a mismatch of configuration in udev daemon.

This issue caused by udev daemon get the hostname and store it in static area.
The spawn(forked) processes shared the same area and get the old hostname. And
DRBD will ignore the configuration that not belong to local host.
```

- Related ticket: https://bugzilla.suse.com/show_bug.cgi?id=1177927
- Needles: N/A
- Verification run: [node 1](http://mango.qa.suse.de/tests/3235) & [node 2](http://mango.qa.suse.de/tests/3236) & [support server](http://mango.qa.suse.de/tests/3234)

### CODE REVIEW

File: tests/ha/iscsi_client.pm [hunk 1]
```diff
@@ -17,14 +17,17 @@ use Utils::Backends qw(is_remote_backend);
...
+use version_utils qw(is_sle);
 
 sub run {
     # Some remote backends connect to the root-console via sshXtermVt or ipmiXtermVt,
     # which set DISPLAY and cause yast2 to show its graphical version. This unsets
     # DISPLAY so the terminal version is shown instead when testing in textmode
     assert_script_run 'unset DISPLAY' if (check_var('DESKTOP', 'textmode') && is_remote_backend());
 
+    # Restart udevd in 15-SP3+ if hostname was changed. See comment 15 in bsc#1177927 for details
+    systemctl 'restart systemd-udevd' if (is_sle('15-SP3+') && get_var('HOSTNAME'));
```
Comment (alvarocarvajald) [inline tests/ha/iscsi_client.pm:29]: I added the fix only for 15-SP3+ as we are currently only seeing the drbd issue in that version, but I also think that a restart of udevd at this point in the test is not really harmful, so we could do it always and for all versions. WDYT @ldevulder @juadk ?

Comment (ldevulder) [reply tests/ha/iscsi_client.pm:29]: Fine for me, and it may appears in post 15-SP3 as well in a maintenance package.

**Review by ldevulder — APPROVED** [review]
LGTM

**Review by ricardobranco777 — APPROVED** [review]
LGTM

File: tests/ha/iscsi_client.pm (hunk 1 above)
Comment (juadk) [reply tests/ha/iscsi_client.pm:29]: Yes, we can do it for all versions.

**Comment by ldevulder** [comment]
In fact after a quick look, maybe it make more sense to add this directly in the `set_hostname` function in `lib/utils.pm`? As there is already restart pf network service in it?
Or we can also consider that it's an issue in `hostnamectl` command or in `udevd`? Maybe the command needs to inform `udevd`, or `udevd` needs to check?

**Review by juadk — APPROVED** [review]

**Comment by juadk** [comment]
> In fact after a quick look, maybe it make more sense to add this directly in the `set_hostname` function in `lib/utils.pm`? As there is already restart pf network service in it?
Yes could be a good idea.

> Or we can also consider that it's an issue in `hostnamectl` command or in `udevd`? Maybe the command needs to inform `udevd`, or `udevd` needs to check?
I don't know what to think about this. Actually, is the udevd restart documented somewhere in case of hostname change? I don't think so.

**Comment by alvarocarvajald** [comment]
> In fact after a quick look, maybe it make more sense to add this directly in the `set_hostname` function in `lib/utils.pm`? As there is already restart pf network service in it?

I thought about placing this in `console/hostname`, but also thought impact and review time could be greater as - like `lib/utils` - it's a module that it's also used outside of HA tests. Also, since it has to do with udevd, figured `ha/iscsi_client` was the better place as it's there we're adding devices to the cluster nodes. But yes, this is tied to the hostname change, so we could place it there instead. I think results will be the same. If you think it's better to move it there, I'll do it. @juadk WDYT?

> Or we can also consider that it's an issue in `hostnamectl` command or in `udevd`? Maybe the command needs to inform `udevd`, or `udevd` needs to check?

Indeed. That's why my follow up [question](https://bugzilla.suse.com/show_bug.cgi?id=1177927#c17) in the bug report was whether we should still consider this a bug or a configuration issue. For the moment, it seems the bug will remain open with a lower priority, seeing as we have a workaround and it's not such a common practice to change the cluster nodes' hostnames before configuring drbd.

**Comment by juadk** [comment]
> I thought about placing this in `console/hostname`, but also thought impact and review time could be greater as - like `lib/utils` - it's a module that it's also used outside of HA tests. Also, since it has to do with udevd, figured `ha/iscsi_client` was the better place as it's there we're adding devices to the cluster nodes. But yes, this is tied to the hostname change, so we could place it there instead. I think results will be the same. If you think it's better to move it there, I'll do it. @juadk WDYT?

And what do you think about placing it in `ha/iscsi_client` in a first time and wait for the bug's conclusion? Then we will decide to move it in a shared piece of code and it does not matter how much time the review will take.

**Comment by ldevulder** [comment]
> Then we will decide to move it in a shared piece of code and it does not matter how much time the review will take.

Sounds a good idea! :-)

File: tests/ha/iscsi_client.pm (hunk 1 above)
Comment (alvarocarvajald) [reply tests/ha/iscsi_client.pm]: > Yes, we can do it for all versions.

Done.

**Pull request merged**